sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -346,6 +348,12 @@ case class FileSourceScanExec(
     } else {
       None
     }
+  } ++ {
+    if (relation.partitionSchemaOption.isDefined) {
+      Some("numPartitions" -> SQLMetrics.createMetric(sparkContext, "number of partitions read"))
```

Comment:
Although previously it is `PartitionCount`, `numPartitions` looks more consistent.